Tests/MapboxDirectionsTests/VisualInstructionComponentTests.swift

```diff
@@ -23,32 +23,126 @@ class VisualInstructionComponentTests: XCTestCase {
+                           alternativeText: .init(text: "I 95", abbreviation: nil, abbreviationPriority: nil),
+                           mapboxShield: .init(baseURL: URL(string: "https://api.mapbox.com/styles/v1/")!, name: "us-interstate", textColor: "black", displayRef: "242"))
+        let encoder = JSONEncoder()
+        var encodedData: Data?
+        XCTAssertNoThrow(encodedData = try encoder.encode(component))
+        XCTAssertNotNil(encodedData)
+        
+        if let encodedData = encodedData {
+            var encodedComponentJSON: [String: Any?]?
+            XCTAssertNoThrow(encodedComponentJSON = try JSONSerialization.jsonObject(with: encodedData, options: []) as? [String: Any?])
+            XCTAssertNotNil(encodedComponentJSON)
+            
+            XCTAssert(JSONSerialization.objectsAreEqual(componentJSON, encodedComponentJSON, approximate: false))
+        }
+    }
+    
+    func testShield() {
+        let shieldJSON = [
+            "base_url": "https://api.mapbox.com/styles/v1/",
+            "name": "us-interstate",
+            "text_color": "black",
+            "display_ref": "242",
+        ]
+        
+        let shieldData = try! JSONSerialization.data(withJSONObject: shieldJSON, options: [])
+        var shield: MapboxShield?
+        XCTAssertNoThrow(shield = try JSONDecoder().decode(MapboxShield.self, from: shieldData))
+        XCTAssertNotNil(shield)
+        let url = URL(string: "https://api.mapbox.com/styles/v1/")
+        if let shield = shield {
+            XCTAssertEqual(shield.baseURL, url)
+            XCTAssertEqual(shield.name, "us-interstate")
+            XCTAssertEqual(shield.textColor, "black")
+            XCTAssertEqual(shield.displayRef, "242")
+        }
+        shield = .init(baseURL: url!, name: "us-interstate", textColor: "black", displayRef: "242")
+        
+        let encoder = JSONEncoder()
+        var encodedData: Data?
+        XCTAssertNoThrow(encodedData = try encoder.encode(shield))
+        XCTAssertNotNil(encodedData)
+        
+        if let encodedData = encodedData {
+            var encodedShieldJSON: [String: Any]?
+            XCTAssertNoThrow(encodedShieldJSON = try JSONSerialization.jsonObject(with: encodedData, options: []) as? [String: Any])
+            XCTAssertNotNil(encodedShieldJSON)
+            
+            XCTAssert(JSONSerialization.objectsAreEqual(shieldJSON, encodedShieldJSON, approximate: false))
+        }
+    }
+    
+    func testShieldImageComponent() {
+        let componentJSON: [String : Any] = [
+            "text": "I 95",
             "type": "icon",
-            "imageBaseURL": "https://s3.amazonaws.com/mapbox/shields/v3/us-42",
+            "mapbox_shield": [
+                "base_url": "https://api.mapbox.com/styles/v1/",
+                "name": "us-interstate",
+                "text_color": "black",
+                "display_ref": "242"
```

Comment:
`text` is for I-95, but `display_ref` is for I-242 and `text_color` is for something other than an Interstate shield. Let’s make sure that this test is internally consistent, referring to the same highway shield throughout, to avoid confusion in the future.

Reply:
I was following a [test](https://github.com/mapbox/mapbox-java/blob/2141e8bdff2dabd01967500ce29fd60be226cf48/services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/BannerComponentTest.java#L165-L180) from mapbox-java. Thanks for this catch - I will make sure it's consistent.